blocks/api/parser.js

```diff
@@ -45,31 +44,150 @@ export function getBlockAttributes( blockNode, blockSettings ) {
 	return attrs;
 }
 
+function htmlEscape( str ) {
+	return str
+		.replace( /&/g, '&amp;' )
+		.replace( /"/g, '&quot;' )
+		.replace( /'/g, '&#39;' )
+		.replace( /</g, '&lt;' )
+		.replace( />/g, '&gt;' );
+}
+
+function htmlUnescape( str ) {
+	return str
+		.replace( /&quot;/g, '"' )
+		.replace( /&#39;/g, '\'' )
+		.replace( /&lt;/g, '<' )
+		.replace( /&gt;/g, '>' )
+		.replace( /&amp;/g, '&' );
+}
+
 /**
  * Returns a list of blocks extracted from the Post Content
  *
  * @param  {String} content The post content
  * @return {Array}          Block list
  */
 export default function parse( content ) {
-	return grammarParse( content ).reduce( ( memo, blockNode ) => {
-		// Use type from block node, otherwise find unknown handler
-		let { blockType = getUnknownTypeHandler() } = blockNode;
-
-		// Try finding settings for known block type, else again fall back
-		let settings = getBlockSettings( blockType );
-		if ( ! settings ) {
-			blockType = getUnknownTypeHandler();
-			settings = getBlockSettings( blockType );
+	// First, convert comment delimiters into temporary <wp-block> "tags" so
+	// that TinyMCE can parse them.  Examples:
+	//   In  : <!-- wp:core/text -->
+	//   Out : <wp-block slug="core/text">
+	//   In  : <!-- /wp:core/text -->
+	//   Out : </wp-block>
+	//   In  : <!-- wp:core/embed url:youtube.com/xxx& -->
+	//   Out : <wp-block slug="core/embed" attributes="url:youtube.com/xxx&amp;">
+	content = content.replace(
+		/<!--\s*(\/?)wp:([a-z0-9/-]+)((?:\s+[a-z0-9_-]+:[^\s]+)*)\s*-->/g,
```

Comment:
Yes - on that front, here is what I would like to see happen next:

- Make it possible to extend TinyMCE's parser to recognize our comment delimiters.  Get rid of this hack to turn them into temporary tags.
- Change the attribute format in our comment delimiters to match regular HTML tags:  `<!-- wp:core/block attr="something with spaces" -->`.  This is more standard and will be much easier for TinyMCE's parser to deal with.